File: java/src/main/java/xyz/k256/sdk/types/Token.java
package xyz.k256.sdk.types;

import java.util.List;
import java.util.Map;
import java.util.Optional;

/**
 * Token metadata.
 */
public record Token(
    /** Token mint address */
    String address,
    /** Token symbol (e.g., "SOL", "USDC") */
    String symbol,
    /** Token name */
    String name,
    /** Token decimals */
    int decimals,
    /** URL to token logo */
    Optional<String> logoUri,
    /** List of tags */
    Optional<List<String>> tags,
    /** Additional metadata */
    Optional<Map<String, Object>> extensions
) {}

File: rust/src/types/token.rs
//! Token types.

use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Token metadata.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Token {
    /// Token mint address
    pub address: String,
    /// Token symbol (e.g., "SOL", "USDC")
    pub symbol: String,
    /// Token name
    pub name: String,
    /// Token decimals
    pub decimals: u8,
    /// URL to token logo
    #[serde(skip_serializing_if = "Option::is_none")]
    pub logo_uri: Option<String>,
    /// List of tags
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    /// Additional metadata
    #[serde(skip_serializing_if = "Option::is_none")]
    pub extensions: Option<HashMap<String, serde_json::Value>>,
}
